addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -137,45 +136,120 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 }
 
                 if (!observedErrors.isEmpty()) {
-                    checkObservedErrors(observedErrors, msg, csp);
+                    checkObservedErrors(observedErrors, msg, csp, false);
                 }
 
                 List<String> allowedWildcardSources = getAllowedWildcardSources(csp);
                 if (!allowedWildcardSources.isEmpty()) {
-                    checkWildcardSources(allowedWildcardSources, msg, csp);
+                    checkWildcardSources(allowedWildcardSources, msg, csp, false);
                 }
 
                 PolicyInOrigin p = new PolicyInOrigin(policy, URI.parseURI(RAND_FQDN).orElse(null));
                 if (p.allowsUnsafeInlineScript()) {
-                    buildScriptUnsafeInlineAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildScriptUnsafeInlineAlert(
+                                    getHeaderField(msg, HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                                            .get(0),
+                                    csp)
                             .raise();
                 }
 
                 if (p.allowsUnsafeInlineStyle()) {
-                    buildStyleUnsafeInlineAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildStyleUnsafeInlineAlert(
+                                    getHeaderField(msg, HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                                            .get(0),
+                                    csp)
                             .raise();
                 }
 
                 if (allowsUnsafeHashes(policy, FetchDirectiveKind.ScriptSrc)) {
-                    buildScriptUnsafeHashAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildScriptUnsafeHashAlert(
+                                    getHeaderField(msg, HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                                            .get(0),
+                                    csp)
                             .raise();
                 }
 
                 if (allowsUnsafeHashes(policy, FetchDirectiveKind.StyleSrc)) {
-                    buildStyleUnsafeHashAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildStyleUnsafeHashAlert(
+                                    getHeaderField(msg, HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                                            .get(0),
+                                    csp)
                             .raise();
                 }
 
                 if (allowsUnsafeEval(policy, FetchDirectiveKind.ScriptSrc)) {
-                    buildScriptUnsafeEvalAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildScriptUnsafeEvalAlert(
+                                    getHeaderField(msg, HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                                            .get(0),
+                                    csp)
```

Comment:
Needs to be handled in meta check and evidence also needs isMeta/getHeaderField ternary handling.